### PR TITLE
support dotfuscator new version

### PR DESCRIPTION
**Pull request by Net7Cracker**
fix stringdecrypt and add control flow cleaner

### CODE REVIEW

File: de4dot.code/deobfuscators/Dotfuscator/StringDecrypter.cs [hunk 1]
```diff
@@ -90,8 +92,20 @@ public class StringDecrypterInfo {
...
-
-					var info = new StringDecrypterInfo(method, ldci4.GetLdcI4Value());
+					var ldarg1 = instrs[i + 4];
+					if (!ldarg1.IsLdarg() || ldarg1.GetParameterIndex() != 1)
+						continue;
+					var opAdd1 = instrs[i + 5];
+					int magicAddInt = 0;
+					bool newVersion = false;
+					if (opAdd1.OpCode == OpCodes.Add) {
+						var ldci4_2 = instrs[i + 6];
```
Comment (0xd4d) [inline de4dot.code/deobfuscators/Dotfuscator/StringDecrypter.cs]: You haven't verified that i + 6 is a valid index here or in the other new code above.

File: de4dot.code/deobfuscators/Dotfuscator/CflowDecrypter.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    class CflowDecrypter {
+		ModuleDefMD module;
+		
+
+		public CflowDecrypter(ModuleDefMD module) {
+			this.module = module;
+		}
+        public void CflowClean()
+        {
+            foreach (TypeDef type in this.module.GetTypes())
```
Comment (0xd4d) [inline de4dot.code/deobfuscators/Dotfuscator/CflowDecrypter.cs]: This whole file is still not formatted. It should use tabs and not spaces. Less important: there are a lot of empty lines for no reason.

Comment (Net7Cracker) [reply de4dot.code/deobfuscators/Dotfuscator/CflowDecrypter.cs]: ok, I will check it.

File: de4dot.code/deobfuscators/Dotfuscator/StringDecrypter.cs (hunk 1 above)
Comment (Net7Cracker) [reply de4dot.code/deobfuscators/Dotfuscator/StringDecrypter.cs]: ok

**Pull request closed without merging**